hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/endpoint/EndpointBase.java

```diff
@@ -98,6 +101,7 @@ protected OzoneBucket getBucket(OzoneVolume volume, String bucketName)
   @PostConstruct
   public void initialization() {
     LOG.debug("S3 access id: {}", s3Auth.getAccessID());
+    ozoneConfiguration.setIfUnset("ozone.om.group.rights", "NONE");
```

Comment:
@kerneltime Where is behavior defined for this configuration?  I cannot find any references in Ozone code.

Reply:
The config is defined in https://github.com/apache/ozone/blob/master/hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/security/acl/OzoneAclConfig.java#L42-L48

Which is used by the Rpc Client. https://github.com/apache/ozone/blob/master/hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java#L221

This is a bit awkward for 2 reasons, the actual string for configuration cannot be searched as plain text as the configuration splits it into 2 parts. Also, the defaults are set deep inside RpcClient vs being set close to the business logic in s3g. Ideally the group security setting should have been part of the Bucket Endpoint processing itself.

Reply:
> The config is defined in https://github.com/apache/ozone/blob/master/hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/security/acl/OzoneAclConfig.java#L42-L48

Thanks, I forgot about that.

> the actual string for configuration cannot be searched as plain text as the configuration splits it into 2 parts.

That's true.

> Also, the defaults are set deep inside RpcClient

Not really, the default is defined in `OzoneAclConfig`, not in `RpcClient`.

> vs being set close to the business logic in s3g. Ideally the group security setting should have been part of the Bucket Endpoint processing itself.

I don't think this setting is specific to S3.

Reply:
The RpcClient invokes the value calculation and sets it to requests which I think is a bit too deep in the logic. RpcClient should be simpler and the decision for ACL defaults should reside closer to the APIs being evaluated.

Reply:
OzoneAclConfig defines the default value as ALL for group access,
```
  @Config(key = "group.rights",
      defaultValue = "ALL",
```

> I don't think this setting is specific to S3.

The default setting is not exclusively for S3 but for all `RpcClient`s as is currently implemented in the `RpcClient`.  For an S3 specific default group privilege, we may need to set it in the the s3gateway (see previous comment to place in the` gateway CDI producer`).